docs/MEMORY-SETTINGS.md

```diff
@@ -0,0 +1,495 @@
+
+All the user applications are running in the sub-cgroup of the `eve-user-apps`
+cgroup. Each user application has its own sub-cgroup.
+
+### Memory settings in EVE
+
+We use explicit memory settings for several of the cgroups. Namely, we set
+memory limits for the `eve`, `eve/containerd`, `pillar`, and all the sub-cgroups
+of the `eve-user-apps` cgroup.
+
+For the EVE-related cgroups, we set the memory limits in the kernel command line
+arguments. For the user applications, we set the memory in runtime using the
+values of RAM.
+
+## EVE-services memory settings
+
+The memory settings for the EVE services are set using the kernel command line
+arguments.
+
+### Kernel command line arguments
+
+The memory settings in EVE are set using kernel command line arguments:
+
+* dom0_mem
+* eve_mem
+* ctrd_mem
+
+#### Format of memory settings
+
+All "<cmd_component_name>_mem" settings have the following format:
+
+```text
+<cmd_component_name>_mem=<soft_limit>,max:<hard_limit>
+```
+
+where:
+
+* `<cmd_component_name>` is one of the following: `dom0`, `eve`, `ctrd`
+* `<soft_limit>` is the soft limit for the memory usage of the cgroup.
+* `<hard_limit>` is the hard limit for the memory usage of the cgroup.
+
+The `<soft_limit>` and `<hard_limit>` are values in human-readable format. They
+can be specified in the following units: `M` (megabytes), `G` (gigabytes). If no
+unit is specified, the value is treated as bytes. For example, `1G` means 1
+gigabyte, `100M` means 100 megabytes, and `1000000` means 1000000 bytes.
+
+If the hard limit is not specified, the soft limit is used as the hard limit.
+
+#### Component to cgroup mapping
+
+The mapping of the `<cmd_component_name>` to the cgroup is not straightforward
+and is misleading. The following table shows the mapping:
+
+| Component   | cgroup            | default value |
+|-------------|-------------------|---------------|
+| `dom0`      | `eve`             | 800M          |
+| `ctrd`      | `eve/containerd/` | 400M          |
+| `eve`       | `eve/services/`   | 650M          |
+| `eve`       | `eve/services/*`  | 650M          |
+
+That way, the `dom0_mem` setting sets the memory limits for the `eve` cgroup,
+the `ctrd_mem` setting sets the memory limits for the `eve/containerd` cgroup,
+and the `eve_mem` setting sets the memory limits for the `eve/services` cgroup
+and all the sub-cgroups of the `eve/services` cgroup.
+
+For example, the following kernel command line arguments:
+
+```text
+dom0_mem=1G,max:2G eve_mem=1000M,max:1200M ctrd_mem=700M,max:800M
+```
+
+will set the soft limit of the `/sys/fs/cgroup/eve/` cgroup to 1G and the hard
+limit to 2G, the soft limit of the `/sys/fs/cgroup/eve/services/` and all the
+sub-cgroups to 1000M and the hard limit to 1200M, and the soft limit of the
+`/sys/fs/cgroup/eve/containerd/` cgroup to 700M and the hard limit to 800M.
+
+#### GRUB override file
+
+These settings can be changed using the `set_global` command in the
+GRUB override file.
```

Comment:
Is this changeable after the device is onboarded?
For example, if user notices that dom0 needs more memory, is he able to change the setting remotely without re-installing the device?

Reply:
Yes, it can be set permanently by mounting the config partition and fixing the grub config.

Mount the configuration directory:
```
eve config mount /mnt
```
Add the desired kernel argument to the GRUB configuration:
```
echo 'set_global dom0_extra_args "<your_kernel_argument>"' >> /mnt/grub.cfg
```
Unmount the configuration directory:
```
eve config unmount
```
Reboot the device:
```
reboot
```

Will add it.